EXT_LaserEnvelope.md

```diff
@@ -0,0 +1,68 @@
+Domain-Specific Naming Conventions for a laser envelope profile
+===============================================================
+
+openPMD extension name: `LaserEnvelope`
+
+
+Introduction
+------------
+
+This extension is specifically designed for the modeling laser beam propagation.
+
+Mesh Based Records (Fields)
+---------------------------
+
+### Naming Conventions for `mesh record`s (field records)
+
+The laser pulse is represented as a single complex field $\mathcal{E}$, describing the envelope of the laser electric field in the paraxial approximation:
+
+```math
+   \begin{aligned}
+   E_x &= \operatorname{Re}\left( \mathcal{E} e^{-i\omega_0t}p_x\right)\\
+   E_y &= \operatorname{Re}\left( \mathcal{E} e^{-i\omega_0t}p_y\right)\\
+   E_z &= 0\\
+   B_x &= -c E_y\\
+   B_y &= c E_x\\
+   B_z &= 0\end{aligned}
+```
+
+where $\operatorname{Re}$ stands for real part,  $E_x$ (resp. $E_y$) is the laser electric field in the x (resp. y) direction, $\mathcal{E}$ is the complex laser envelope described in the standard, $\omega_0 = 2\pi c/\lambda_0$ is the angular frequency defined from the laser wavelength $\lambda_0$ and $(p_x,p_y)$ is the (complex and normalized) polarization vector. The polarization state (linear, circular, elliptical) is controlled by the phase of the polarization vector. For instance, if $arg(p_x) = arg(p_y)$, the polarization is linear. If $arg(p_x) = arg(p_y) + \pi/2$, the polarization is circular.
+
+When added to an output, the following naming conventions shall be used for complex electric field `mesh records`.
+
+- `laser_envelope/`
+  - type: *(complexX)*
+  - scope: *(required)*
+  - decription: Scalar field for the envelope (in V/m). See above for description.
+  - unitDimension = `(1., 1., -3., -1., 0., 0., 0.)` $(V/m = kg . m / (A . s^3))$
+
+### Additional attributes on the `mesh record` named `electricField`
+
+On the `series` object, set the following attributes:
+
+  - `angularFrequency`
+    - type: *(floatX)*
+    - description: Angular frequency $\omega_0$ at which the laser envelope is defined.
+    - scope: *required*
+    - `unitDimension = (0., 0., -1., 0., 0., 0., 0.)` (rad/s)
+
+
+  - `polarization`
+    - type: *(floatX)*
+    - description: Polarization vector. This vector is normalized (so its modulus is 1) and contains all information to account for polarization states and carrier-envelope phase (CEP).
+    - scope: *required*
+    - components:
+      - `x/`
+      - `y/`
+    - `unitDimension = (0., 0., 0., 0., 0., 0., 0.)` (unitless)
+
+
+  - `propagationDirection`
+    - type: *(floatX)*
+    - scope: *optional*
+    - description: Propagation direction for the laser pulse. Default is (0,0,1), i.e. propagation along the z coordinate.
+    - components:
+      - `x/`
+      - `y/`
+      - `z/`
+    - `unitDimension = (1., 0., 0., 0., 0., 0., 0.)` (length)
```

Comment:
```suggestion
```

I am not sure that this attribute is actually needed. I think it would enough to specify all the other parameters, and the propagation direction would be chosen by the user in their simulation input script, independently from this file.

Reply:
The goal of this optional parameter is to address the practical case when this file would be shared between different people to run the same simulation with e.g. different codes. There it could be valuable to make sure all the required laser information is contained in the openPMD file, rather than sharing the file + some information like the propagation direction. In practice, this could be used e.g. to set the antenna normal vector. Options are:
- Drop this idea, and also communicated the propagation direction (and potentially other elements that I missed)
- Keep this parameter (and fix the array issue, similar to above)
- Include it in a different way in the array (although I don't see a simple and obvious way)

What do you think?

Reply:
Hm, I don't have a very strong opinion, but I guess for now it might be okay to just drop this parameter.
We could also discuss this point further at the next `lasy` meeting.

Reply:
Sounds good, let's remove it, then we can add it again if needed